release/models/mpls/openconfig-mpls-types.yang

```diff
@@ -459,6 +466,7 @@ module openconfig-mpls-types {
             labels RFC 6790";
         }
         enum NO_LABEL {
+          value -1;
```

Comment:
That's correct - if the design pattern is assigning values to variants to match an underlying spec and some are omitted (possibly because there is a mix of spec and _other_) then they are left to the auto assignment rules per RFC and thus this has an issue for any future expansion

Reply:
Is this, in reality, a problem? In what cases are we serialising these enumerations according to their `value` specified here? XML and JSON serialisations only use the string variant, as does gNMI.

Honestly, I feel like the better design pattern is to remove the values from the other cases, but I'd like to understand the overall motivation here.

Reply:
I mention this in my other comment but the wire protocol is not affected in any cases (that I know of today) - in reality, there could be an encoding scheme that carries the value instead of the string to save a few bytes on the wire but as of today the only "issues" are backend related and custom to the implementation.

So this is to spark a consistent design pattern and don't have that strong opinion on either way - statically assign all or none.  If statically assigning all then what benefit does this provide for all cases.

Reply:
IIRC, the CBOR encoding of YANG uses the enum values (except in unions, where the string form is used), so would be impacted by this change.  Of course, it is unclear whether the CBOR encoding would ever expect be used with OpenConfig YANG models.

Reply:
It seems if code generation is to yang spec, it doesn't really matter if there are or are not values for every enum.  But why mix these definitions?

IMO, it's better to have all items within a given enumeration with a value, or all without in order to get consistent, deterministic behavior/readability.  But I guess there is not really a problem with mixing these definitions so long as client and server are generating code off the same oc-model version.

Reply:
@dplore -- not quite sure i understand the comment here -- what does "code generation is to yang spec" mean?

Reply:
I guess if there are no encodings that carry the value of an enum, then this thread is irrelevant.   But to explain my possibly misinformed or erroneous comment, if there were some encoding that carried the value  (instead of the string) and that encoding was performed by a class that was generated from this yang per RFC, then the values would be correct, even if some enums specify them and some do not and even if the order changes.    If the value was set by a class that was not generated from the yang (but written by hand for example), then changing the order of the definition of the enums where some have specified values and others don't would change the value assigned to the unspecified enums which could become a difficult to track down bug.   (Again which is all irrelevant if there is no encoding of the values as was already pointed out)

Reply:
I am not familiar with all code generation or encodings, we use enumerated values in ygot's generated Go, but the values should be opaque to the actual underlying value as they are specified by name rather than by integer. gNMI's encoding works based on the string.

I'm OK to get rid of them - and take the major revision update, I was just questioning whether this was something that needs to create churn in the models (it seems ~relatively harmless to have them today).

Reply:
I think until there is an encoding that uses the value directly, we are safe to change what we want and pick a direction for client/server compatibility.  This can obviously have impacts to any projects out there using OpenConfig models with alternate transports and encodings.

Anything else is implementation detail that can be handled at this time imo (if we are talking within the confines of OC projects).

If an encoding is introduced at a later date that utilizes these values directly then we are faced w/ following the ordering rules or reintroducing explicit values but the partial definitions I would steer clear from imo - essentially back to all or nothing.

Since this also has impact to PR https://github.com/openconfig/public/pull/673, I am fine removing _all_ variant values within these PRs if we are in agreement this is the consistent path fwd for now.  This can always be subject to change later on...